apollo-router/src/plugins/traffic_shaping/mod.rs

```diff
@@ -273,8 +274,8 @@ impl Plugin for TrafficShaping {
     }
 }
 
-pub(crate) type TrafficShapingSubgraphFuture<S> = Either<
-    Either<
+pub(crate) type TrafficShapingSubgraphFuture<S> = EitherResponseFuture<
+    EitherResponseFuture<
         BoxFuture<'static, Result<subgraph::Response, BoxError>>,
         BoxFuture<'static, Result<subgraph::Response, BoxError>>,
```

Comment:
Hm, that's funny, these branches are the same so this probably doesn't need to be an `Either` anymore.